src/admin/class-importer.php

```diff
@@ -157,10 +159,33 @@ private static function import_geo_file( $post_id, $meta_key, $meta_value ) {
 	 * @return void
 	 */
 	public static function import_locations( $post_id, $meta_value = false ) {
-		$file = get_attached_file( get_post_meta( $post_id, 'datalayer_file_id', true ) );
-		// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- We need to read the file contents.
-		$data       = file_get_contents( $file );
-		$geom       = geoPHP::load( $data );
+		$datalayer_type = get_post_meta( $post_id, 'datalayer_type', true );
+
+		switch ( $datalayer_type ) {
+			case 'fileinput':
+			default:
+				$file = get_attached_file( get_post_meta( $post_id, 'datalayer_file_id', true ) );
+				// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- We need to read the file contents.
+				$data = file_get_contents( $file );
+				break;
+			case 'url':
+				$data = Datalayers::fetch_datalayer_url_data( $post_id );
+
+				// Convert data to GeoJSON.
+				$data = Helper::array_to_geojson( $data );
```

Comment:
Waarom is deze stap nodig? De URL zou toch al geodata moeten zijn en door geoPHP te gebruiken kun je er voor zorgen dat het ook terugkomt als geoJSON

Reply:
Heb het aangepast naar geoPHP::load met ook een try en catch om af te vangen of het bestand goed ingeladen kan worden